### DISCUSSION TITLE
How to set predefined trials?

### DISCUSSION THREAD

**Discussion by vl-dud** [optuna/optuna#5186] [Q&A] [answered]
Let's say I have initial assumptions regarding optimal hyperparameters. So I use the code:
```
import optuna

sampler = optuna.samplers.TPESampler(seed=42)
study = optuna.create_study(direction="minimize", sampler=sampler)
distributions = {
	"x": optuna.distributions.IntDistribution(1, 10),
	"y": optuna.distributions.IntDistribution(1, 10)
}
predefined_params = [
	{
		"x": 1,
		"y": 2
	},
	{
		"x": 3,
		"y": 4
	}
]
for params in predefined_params:
	trial = optuna.trial.create_trial(
		state=optuna.trial.TrialState.WAITING,
		params=params,
		distributions={k: v for k, v in distributions.items() if k in params}
	)
	study.add_trial(trial)
	
def objective(trial):
	# Some objective
	
study.optimize(objective, n_trials=100)
```
to add two predefined trials to the study. I suggest that two first trials will be defined by me, and the rest will be random from the search space. Will this work correctly?  Did I choose a right way to solve the problem? Maybe it would be better to customize the `sampler` or use `FixedTrial` objects?

**Comment by nzw0301** [comment, marked as answer]
[`Study.enqueue_trial`](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.study.Study.html#optuna.study.Study.enqueue_trial) is all you need.

**Reply by vl-dud** [reply]
Thank you!
